web/api/web_api_v1.c

```diff
@@ -195,14 +195,19 @@ inline uint32_t web_client_api_request_v1_data_google_format(char *name) {
 
 
 inline int web_client_api_request_v1_alarms(RRDHOST *host, struct web_client *w, char *url) {
+    (void)url;
     int all = 0;
 
-    while(url) {
-        char *value = mystrsep(&url, "&");
-        if (!value || !*value) continue;
+    uint32_t end = w->total_params;
+    if(end) {
+        uint32_t  i = 0;
+        do {
+            char *value = w->param_values[i].body;
+            size_t length = w->param_values[i].length;
 
-        if(!strcmp(value, "all")) all = 1;
-        else if(!strcmp(value, "active")) all = 0;
+            if(!strncmp(value, "all",length)) all = 1;
```

Comment:
Something doesn't work with this parameter. When you send https://localhost:19999/api/v1/alarms?all, you're supposed to get all the configured alarms. Instead, we get all the raised alarms, i.e. exactly the same as if we don't pass "all" at all.

Reply:
I fixed the error, there was a bug in parser this kind of query strings.

Reply:
Not fixed completely. Now the following doesn't work properly:
http://localhost:19999/api/v1/alarms?all&132111